api/blockindexer/indexer.go

```diff
@@ -0,0 +1,130 @@
+// Copyright (C) 2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package txindexer
+
+import (
+	"sync"
+
+	"github.com/ava-labs/avalanchego/ids"
+
+	"github.com/ava-labs/hypersdk/chain"
+	"github.com/ava-labs/hypersdk/event"
+	"github.com/ava-labs/hypersdk/internal/cache"
+	"github.com/ava-labs/hypersdk/vm"
+)
+
+const (
+	Name      = "blockindexer"
+	Namespace = "blockindexer"
+)
+
+var (
+	_ event.SubscriptionFactory[*chain.StatefulBlock] = (*subscriptionFactory)(nil)
+	_ event.Subscription[*chain.StatefulBlock]        = (*indexer)(nil)
+)
+
+type Config struct {
+	Enabled             bool `json:"enabled"`
+	AcceptedBlockWindow int  `json:"acceptedBlockWindow"`
+}
+
+func NewDefaultConfig() Config {
```

Comment:
nit: call this `NewConfig`?

Reply:
Following the existing style here. I do prefer this as is though, `NewDefaultConfig` means no parameters all default values to me and `NewConfig` would take in parameters imo.

We only need `NewDefaultConfig` (at least for now) since we construct the actual config by creating the default and unmarshalling JSON into it.